src/Functions/CMakeLists.txt

```diff
@@ -3,6 +3,31 @@ add_subdirectory(divide)
 include("${ClickHouse_SOURCE_DIR}/cmake/dbms_glob_sources.cmake")
 add_headers_and_sources(clickhouse_functions .)
 
+# These files are moved from clickhouse_functions into dbms to make dbms self-contained (not depend on clickhouse_functions)
+# This allows less dependency and linker work (specially important when building many example executables)
+set(DBMS_FUNCTIONS
+    IFunction.cpp
+    FunctionFactory.cpp
+    FunctionHelpers.cpp
+    extractTimeZoneFromFunctionArguments.cpp
+    FunctionsLogical.cpp
+    if.cpp
+    multiIf.cpp
+    multiMatchAny.cpp
+    checkHyperscanRegexp.cpp
+    CastOverloadResolver.cpp
+    FunctionsConversion.cpp # createFunctionBaseCast
```

Comment:
Why does `dbms` require these functions implementations?

Reply:
Depends, but usually because it tries to instantiate a specific function in a pass / AST visitor. I recovered the list from https://github.com/ClickHouse/ClickHouse/pull/65705, but I'll try to add comments to each file.